lua/entities/gmod_wire_expression2/core/custom/constraintcore.lua

```diff
@@ -166,6 +166,22 @@ e2function void rope(index, entity ent1, vector v1, entity ent2, vector v2, addl
 	addundo(self, ent1.data.Ropes[index], "rope")
 end
 
+--- Creates a rope between <ent1> and <ent2> at vector positions local to each ent, with <addlength> additional length, <width> width, and <mat> material.
+e2function void rope(index, entity ent1, vector v1, entity ent2, vector v2, addlength, width, string mat, rigid )
+	if not checkEnts(self, ent1, ent2) then return end
+	if not ent1.data then ent1.data = {} end
+	if not ent1.data.Ropes then ent1.data.Ropes = {} end
+	local vec1, vec2 = Vector(v1[1], v1[2], v1[3]), Vector(v2[1], v2[2], v2[3])
+	local length = (ent1:LocalToWorld(vec1) - ent2:LocalToWorld(vec2)):Length()
+	
+	if IsValid(ent1.data.Ropes[index]) then 
```

Comment:
Factoring some of this out into a helper function would be nice, in order to reduce copy-pasted errors.